components/SortMenu.tsx

```diff
@@ -0,0 +1,95 @@
+import React, { useState } from "react";
+import { Menu, Button, Typography, Icon } from "@equinor/eds-core-react";
+import { chevron_down } from "@equinor/eds-icons";
+import { useRouter } from "next/router";
+
+export default function SortMenu(): JSX.Element {
+  const router = useRouter();
+  const path = router.pathname;
+  const { orderBy } = router.query;
+
+  const buttonLabelDictionary = {
+    name: "Alphabetically",
+    created: "Last created",
+    modified: "Last modified",
+  };
+
+  const [state, setState] = useState<{
+    buttonEl: HTMLButtonElement;
+  }>({ buttonEl: null });
+
+  const { buttonEl } = state;
+  const isOpen = Boolean(buttonEl);
+
+  const openMenu = (
+    e:
+      | React.MouseEvent<HTMLButtonElement, MouseEvent>
+      | React.KeyboardEvent<HTMLButtonElement>
+  ) => {
+    const target = e.target as HTMLButtonElement;
+    setState({ ...state, buttonEl: target });
+  };
+
+  const closeMenu = () => setState({ ...state, buttonEl: null });
+
+  const handleClick = (newOrderBy: string) =>
+    router.push({ pathname: path, query: { orderBy: newOrderBy } });
+
+  return (
+    <div
+      style={{
+        display: "flex",
+        flexDirection: "row",
+        justifyContent: "space-around",
+        alignItems: "center",
+      }}
+    >
+      <Typography variant="h6">
+        {buttonLabelDictionary[orderBy.toString()]}
```

Comment:
smells like python 🐍

Reply:
I guess if the lookup is undefined the Typography component displays nothing without a crash?

Reply:
Lol it does, now that you mention it 😆

Reply:
Also I have tested with an undefined lookup and the Typography does not crash, it is simply empty.